packages/evm/src/evm.ts

```diff
@@ -858,6 +858,9 @@ export class EVM implements EVMInterface {
       } else {
         message.code = await this.eei.getContractCode(message.codeAddress)
         message.isCompiled = false
+        if (this._common.isActivatedEIP(3540)) {
+          message.code = getEOFCode(message.code)
```

Comment:
Do we really need this?  `runInterpreter` checks the code passed in as part of the message for the EOF header and sets the PC to the correct position based on the sections header.

Reply:
I see that more tests fail when I comment it out but from how we handle code in the interpreter, it seems like they shouldn't be failing.  I looked back at the original PR and the idea was that we handled all the bits about defining the code section inside `interpreter.run` so this theoretically should be duplicative.